[iOS] Update GPU sandbox to match macOS behavior

brentfulgham opened the pull request:
#### b069ece7b5e303fcd1010fadb197164a855f64a8
<pre>
[iOS] Update GPU sandbox to match macOS behavior
<a href="https://bugs.webkit.org/show_bug.cgi?id=276498">https://bugs.webkit.org/show_bug.cgi?id=276498</a>
&lt;<a href="https://rdar.apple.com/problem/131556690">rdar://problem/131556690</a>&gt;

Reviewed by Per Arne Vollan.

The iOS GPU sandbox denies access to a few sysctl calls that are allowed on macOS,
and (more importantly) are allowed in the iOS WebContent process. These are showing
up in telemetry when performing media and other web page loads, and are likely to be
impacting performance.

We should allow these sysctl-read calls, as well as a MIG routine needed for proper
memory attribution.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.GPU.sb.in:

Canonical link: <a href="https://commits.webkit.org/280924@main">https://commits.webkit.org/280924@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d060e9a4b17636cf0147b43fe15622cf4f23cddb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/57904 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/37232 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/10380 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/61526 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/8349 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/44868 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/8537 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/46924 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/5941 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/59934 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/34917 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/50044 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/27750 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/31683 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/7353 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/53631 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/7607 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/63212 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/1818 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/7679 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/54147 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/1824 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/50055 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/54280 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/1549 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8656 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/33061 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/34147 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/35231 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/33892 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pvollan commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.GPU.sb.in:
```diff
@@ -585,7 +588,9 @@
         "vm.footprint_suspend"
-        "vm.malloc_ranges")) ;; <rdar://problem/105161083>
+        "vm.malloc_ranges") ;; <rdar://problem/105161083>
+    (sysctl-name-prefix "hw.perflevel") ;; <rdar://problem/76782530>
```
I think it would be good to avoid prefix notation, since this can potentially allow more than needed.

brentfulgham replied:
That's a good point. However, I think we should use the prefix notation for these `hw.perflevel` things since we are already using this syntax for the WebContent sandbox on all platforms (plus the GPU Process on macOS), and that was based on the rules SEAR approved for the other OS sandboxes.

pvollan reviewed (approved):
R=me.

brentfulgham committed:
[iOS] Update GPU sandbox to match macOS behavior
https://bugs.webkit.org/show_bug.cgi?id=276498
<rdar://problem/131556690>

Reviewed by Per Arne Vollan.

The iOS GPU sandbox denies access to a few sysctl calls that are allowed on macOS,
and (more importantly) are allowed in the iOS WebContent process. These are showing
up in telemetry when performing media and other web page loads, and are likely to be
impacting performance.

We should allow these sysctl-read calls, as well as a MIG routine needed for proper
memory attribution.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.GPU.sb.in:

Canonical link: https://commits.webkit.org/280924@main
